lib/node/runner/checker_ballot_transaction.go

```diff
@@ -244,14 +254,79 @@ func ValidateOp(st *storage.LevelDBBackend, source *block.BlockAccount, op trans
 			err = errors.ErrorFrozenAccountNoDeposit
 			return
 		}
-		// If it's a frozen account, everything must be withdrawn
 		if source.Linked != "" {
+			// Source account's linked address must be equal to target address
+			if source.Linked != casted.TargetAddress() {
+				err = errors.ErrorUnfreezingToInvalidLinkedAccount
```

Comment:
Why ?

Reply:
@Geod24 I think freezing and unfreezing should be executed only through linked general account.
Because I think frozen account are like as installment savings.
So isn't it natural to use freezing and unfreezing through only one account?

Reply:
I don't think we should put that restriction in place. We should only care about things that are definitely wrong, not trying to install policies. If account A1 decides to make frozen account F1, which rewards A2, and then to give the key to A3, that's his decision.